watch crash when trying to post unreadable file.

![image](https://github.com/user-attachments/assets/803bb089-6a48-42df-acd7-8dfa1163dcf0)

snapshot is for v2, but I think the same will happen with v3.  need to handle failure to open file for checksumming.

reproduced on sr3:
```
2024-11-19 16:54:57,399 [ERROR] sarracenia.flow gather flowCallback plugin <bound method File.gather of <sarracenia.flowcb.gather.file.File object at 0x7f051f125480>> crashed: [Errno 13] Permission denied: '/home/peter/sarra_devdocroot/downloaded_by_sub_amqp/unreadable'
2024-11-19 16:54:57,399 [DEBUG] sarracenia.flow gather details:
Traceback (most recent call last):
  File "/home/peter/sr3/sarracenia/flow/__init__.py", line 1123, in gather
    retval = p(self.o.batch-so_far)
  File "/home/peter/sr3/sarracenia/flowcb/gather/file.py", line 726, in gather
    messages.extend(self.watch_dir(d))
  File "/home/peter/sr3/sarracenia/flowcb/gather/file.py", line 670, in watch_dir
    return self.walk(sld)
  File "/home/peter/sr3/sarracenia/flowcb/gather/file.py", line 577, in walk
    messages.extend(self.post1file(path, sarracenia.stat(path)))
  File "/home/peter/sr3/sarracenia/flowcb/gather/file.py", line 397, in post1file
    messages.extend(self.post_file(path, lstat))
  File "/home/peter/sr3/sarracenia/flowcb/gather/file.py", line 157, in post_file
    msg = sarracenia.Message.fromFileData(path, self.o, lstat)
  File "/home/peter/sr3/sarracenia/__init__.py", line 666, in fromFileData
    m.computeIdentity(path, o)
  File "/home/peter/sr3/sarracenia/__init__.py", line 520, in computeIdentity
    fp = open(path, 'rb')
PermissionError: [Errno 13] Permission denied: '/home/peter/sarra_devdocroot/downloaded_by_sub_amqp/unreadable'
```

--- a/sarracenia/__init__.py
+++ b/sarracenia/__init__.py
@@ -662,12 +662,17 @@ class Message(dict):
     def fromFileData(path, o, lstat=None):
         """
             create a message based on a given file, calculating the checksum.
-            returns a well-formed message, or none.
+            returns a well-formed message, or None.
         """
         m = sarracenia.Message.fromFileInfo(path, o, lstat)
         if lstat :
             if os_stat.S_ISREG(lstat.st_mode):
-                m.computeIdentity(path, o)
+                try:
+                    m.computeIdentity(path, o)
+                except Exception as ex:
+                    logger.error( f" failed to identify {path}: {ex} ")
+                    return None
+
                 if features['filetypes']['present']:
                     try:
                         t = magic.from_file(path,mime=True)

--- a/sarracenia/flow/__init__.py
+++ b/sarracenia/flow/__init__.py
@@ -956,7 +956,7 @@ class Flow:
                         if msg['fileOp'][f].startswith(u.path):
                             msg['fileOp'][f] = msg['fileOp'][f].replace(u.path, new_dir, 1)
                             
-        if self.o.mirror and len(token) > 1:
+        if mirror and len(token) > 1:
             new_dir = new_dir + '/' + '/'.join(token[:-1])
 
         new_dir = self.o.variableExpansion(new_dir, msg)

--- a/sarracenia/flow/__init__.py
+++ b/sarracenia/flow/__init__.py
@@ -1010,7 +1010,7 @@ class Flow:
                     self.reject( m, 410, f"file too old: {age:g} sec. skipping: {m.getIDStr()}, ")
                     continue
 
-                if self.o.fileAgeMin > 0 and age < self.o.fileAgeMin:
+                if self.o.component != 'poll' and self.o.fileAgeMin > 0 and age < self.o.fileAgeMin:
                     logger.warning( f"file too young: queueing for retry.")
                     self.worklist.failed.append(m)
                     continue

--- a/sarracenia/flow/__init__.py
+++ b/sarracenia/flow/__init__.py
@@ -1477,7 +1477,16 @@ class Flow:
         """
         # assert
 
-        lstat = sarracenia.stat(msg['new_path'])
+        try:
+            lstat = sarracenia.stat(msg['new_path'])
+        except PermissionError: 
+            # file is gone, so must obtain again?... if it's EPERM... this is a loop... hmm...
+            logger.error( f" file cannot be overwritten {msg['new_path']}: {ex}" )
+            return False
+        except Exception as ex:
+            logger.error( f" likely race condition, failed to stat a second time {msg['new_path']}: {ex}" )
+            return True
+
         fsiz = lstat.st_size
 
         # FIXME... local_offset... offset within the local file... partitioned... who knows?

--- a/sarracenia/flowcb/gather/file.py
+++ b/sarracenia/flowcb/gather/file.py
@@ -156,6 +156,9 @@ class File(FlowCB):
 
         msg = sarracenia.Message.fromFileData(path, self.o, lstat)
 
+        if not msg: # failed to create message
+            return []
+
         # used when moving a file
         if key != None:
             if not 'fileOp' in msg:
@@ -254,7 +257,12 @@ class File(FlowCB):
             msg['size']=length
 
             # set partstr
-            msg.computeIdentity(path, self.o, offset=offset )
+            try:
+                msg.computeIdentity(path, self.o, offset=offset )
+            except Exception as ex:
+                logger.error( f"could not identify {path}: {ex}" )
+                return []
+
             msg['blocks']['manifest'][current_block] = { 'size':length, 'identity': msg['identity']['value'] }
 
         

